### DISCUSSION TITLE
Experience with production deployment JuiceFS

### DISCUSSION THREAD

**Discussion by chudyandrej** [juicedata/juicefs#2378] [Q&A]
Hi, JuiceFS community. I'm Andrej infrastructure engineer in the [Deepnote](https://deepnote.com/). We considering replacing our ZFS storage setup with this shiny product. So far we made some tests and we have a very good feeling about the product. But I would like to ask if someone here uses JuiceFS in the **production environment**. I would like to know some real-world **experience/limitations** from the production environment. We are planning to deploy JuiceFS to our Kubernetes cluster. We expect millions of top-level directories that will represent mountable userspace. We expect a storage size of hundreds of TB. Does someone run JuiceFS with a similar size?

**Comment by xiaogaozi** [comment]
You can see all JuiceFS adopters [here](https://github.com/juicedata/juicefs/blob/main/ADOPTERS.md), there are already many companies using JuiceFS in production environment.

**Reply by xiaogaozi** [reply]
The storage size of some adopters has reached the PB level.
